build.sh

```diff
@@ -1,12 +1,42 @@
 #!/bin/bash
-set -eo pipefail
+set -eox pipefail
 
-if [ -z "$1" ] ; then
-  echo "Usage: ./build.sh \$VERSION"
+if [ -z "$1" ]; then
+  echo "Usage: ./build.sh \$VERSION_SHORT"
 else
-  VERSION="$1"
+  VERSION_SHORT="$1"
 fi
 
-cd "$VERSION" || exit
+VERSION=$(grep -oP '[0-9]+\.[0-9]+\.[0-9]+' "$VERSION_SHORT/Dockerfile" | head -1)
+DOCKER_REPO=factoriotools/docker_factorio_server
+cd "$VERSION_SHORT" || exit
 
-docker build . -t "factoriotools/docker_factorio_server:$VERSION"
+if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then
+  if [ "$TRAVIS_BRANCH" == "master" ]; then
+    TAG="$VERSION -t $DOCKER_REPO:$VERSION_SHORT"
```

Comment:
When we do changes on master and the tag didn't change, it will rebuild the tag.
We should only build on tag changes for master branch or?

Reply:
We should always build the docker image to check if it is building at all and only push it when we tag it.